### PR TITLE
Replace Name with Description

### LINKED ISSUE #39: Replace Name with Description

### PR DESCRIPTION
**Pull request by bchenghi**
Close #39 
- Rename Name and NameContainsKeywordsPredicate to Description and DescriptionContainsKeyWordsPredicate
- Update CLISyntax to d/
- Validation Regex accepts apostrophe (e.g. Lay's)
- Replace name and its contents in json files to description and description of foods respectively (addressbook.json, duplicatePersonAddressBook.json and typicalPersonsAddressBook.json)
- Update MESSAGE_USAGE in Add, Edit and Find commands
- Update Person objects in TypicalPersons and SampleDataUtil to have food descriptions instead of names of people
- Update CommandTestUtil, AddCommandTest, DescriptionTest and DescriptionContainsKeyWordsPredicateTest with food description strings instead of names of people

### CODE REVIEW

File: src/test/data/JsonAddressBookStorageTest/invalidPersonAddressBook.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "persons": [ {
-    "name": "Person with invalid name field: Ha!ns Mu@ster",
+    "description": "Person with invalid name field: Ha!ns Mu@ster",
```
Comment (sagarsureka) [inline src/test/data/JsonAddressBookStorageTest/invalidPersonAddressBook.json]: Should this be invalid description field?

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 2]
```diff
@@ -48,7 +48,8 @@
     public static final String TAG_DESC_FRIEND = " " + PREFIX_TAG + VALID_TAG_FRIEND;
     public static final String TAG_DESC_HUSBAND = " " + PREFIX_TAG + VALID_TAG_HUSBAND;
 
-    public static final String INVALID_NAME_DESC = " " + PREFIX_NAME + "James&"; // '&' not allowed in names
+    public static final String INVALID_DESCRIPTION_DESC = " " + PREFIX_DESCRIPTION
+            + "James&"; // '&' not allowed in names
```
Comment (sagarsureka) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java]: Not a big issue but do you think we should change names to descriptions in the comment?

File: src/test/java/seedu/address/model/person/PersonTest.java [hunk 3]
```diff
@@ -26,70 +26,70 @@ public void asObservableList_modifyList_throwsUnsupportedOperationException() {
...
-        assertFalse(ALICE.isSamePerson(null));
+        assertFalse(ANCHOVIES.isSamePerson(null));
 
         // different phone and email -> returns false
-        Person editedAlice = new PersonBuilder(ALICE).withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_BOB).build();
-        assertFalse(ALICE.isSamePerson(editedAlice));
+        Person editedAlice = new PersonBuilder(ANCHOVIES).withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_BOB).build();
+        assertFalse(ANCHOVIES.isSamePerson(editedAlice));
 
         // different name -> returns false
```
Comment (sagarsureka) [inline src/test/java/seedu/address/model/person/PersonTest.java]: Maybe the occurrences of name in comments should be replaced with description?

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java [hunk 4]
```diff
@@ -13,87 +13,89 @@
...
     @Test
     public void toModelType_validPersonDetails_returnsPerson() throws Exception {
-        JsonAdaptedPerson person = new JsonAdaptedPerson(BENSON);
-        assertEquals(BENSON, person.toModelType());
+        JsonAdaptedPerson person = new JsonAdaptedPerson(BAGEL);
+        assertEquals(BAGEL, person.toModelType());
     }
 
     @Test
     public void toModelType_invalidName_throwsIllegalValueException() {
```
Comment (sagarsureka) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java]: Maybe we can replace Name in method name with Description, here and in other occurrences?

**Review by sagarsureka — COMMENTED** [review]
Great job! Just a few minor changes in the occurrences of "name", mostly in comments.

File: src/test/data/JsonAddressBookStorageTest/invalidPersonAddressBook.json (hunk 1 above)
Comment (bchenghi) [reply src/test/data/JsonAddressBookStorageTest/invalidPersonAddressBook.json]: Okay changed

File: src/test/java/seedu/address/model/person/PersonTest.java (hunk 3 above)
Comment (bchenghi) [reply src/test/java/seedu/address/model/person/PersonTest.java]: Changed all comments with name to description.

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java (hunk 4 above)
Comment (bchenghi) [reply src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java]: I agree, changed them. Thanks!

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java (hunk 2 above)
Comment (bchenghi) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java]: Yeap I agree. Changed them.

**Review by sagarsureka — APPROVED** [review]
Thank you! LGTM

**Review by jonahtanjz — APPROVED** [review]
LGTM

**Pull request merged**